libsolidity/interface/FileReader.h

```diff
@@ -0,0 +1,87 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <boost/filesystem.hpp>
+#include <map>
+#include <vector>
+
+#include <libsolidity/interface/ReadFile.h>
+
+namespace solidity {
+
+class FileReader
```

Comment:
Can you extract this as its own pull request? @cameel might make some modifications here, soon.

Reply:
Yeah a separate PR would be a good idea. Fortunately I'm only refactoring the path resolution (mapping, allowed paths, base path), not the reader itself and I see that this logic has been moved from the CLI almost unchanged so joining our code should not be hard. It'll be just matter of replacing these individual parts with a single object that handles the resolution.

Reply:
oh why did i see that just now. sure i will (now)... ;(